save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt

```diff
@@ -48,18 +48,30 @@ class TestDiscoveringService {
      * @return a list of [TestSuiteDto]s
      * @throws IllegalArgumentException when provided path doesn't point to a valid config file
      */
+    @Suppress("UnsafeCallOnNullableType")
     fun getAllTestSuites(
         project: Project?,
         rootTestConfig: TestConfig,
         propertiesRelativePath: String,
         testSuiteRepoUrl: String) = rootTestConfig
         .getAllTestConfigs()
-        .mapNotNull { it.getGeneralConfigOrNull()?.suiteName }
-        .map { suiteName ->
+        .asSequence()
+        .mapNotNull { it.getGeneralConfigOrNull() }
+        .filterNot { it.suiteName == null }
```

Comment:
Good check, but do we have suites with null name?

Reply:
Maybe in topmost configs, that already have general section, but expect suite names to be filled in child configs?